# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/bin
install fbcat $DESTDIR/fb/bin

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/fbcat-build.pc << EOF
Name: fbcat
Version: 0.5.2
Description: Linux Framebuffer screenshot grabber
Requires: $REQUIRES

exec=\\
/fb/bin/fbcat
EOF

exit

# patch
--- fbcat-0.5.2.orig/Configure
+++ fbcat-0.5.2/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile

# source
https://github.com/jwilk/fbcat/archive/0.5.2/fbcat-0.5.2.tar.gz
